ex7.3_post_office2.py

```diff
@@ -0,0 +1,124 @@
+                                            Urgent messages appear first.
+                Returns:
+                    int: The message ID, auto incremented number.
+                Raises:
+                    KeyError: If the recipient does not exist.
+                Examples:
+                    After creating a PO box and sending a letter,
+                    the recipient should have 1 message in the
+                    inbox.
+                    >>> po_box = PostOffice(['a', 'b'])
+                    >>> message_id = po_box.send_message('a', 'b', 'Hello!')
+                    >>> len(po_box.boxes['b'])
+                    1
+                    >>> message_id
+                    1
+                """
+        user_box = self.boxes[recipient]
+        self.message_id = self.message_id + 1
+        message_details = {
+            'id': self.message_id,
+            'body': message_body,
+            'sender': sender,
+        }
+        if urgent:
+            user_box.insert(0, message_details)
+        else:
+            user_box.append(message_details)
+        return self.message_id
+
+    def read_inbox(self, username, N=10):
+        """
+        An function that gets the name of the user and number of the message request.
+        the function returns the first N messages in the mailbox.
+        :param username: name of the user
+        :param N: number of the message request
+        :return: the first N messages in the mailbox. (if not exist return all messages in the mailbox)
+        """
+        return_message = []
+        count_message = 0
+        if not N == 0:
+            if N < len(self.boxes[username]):
+                for message in range(len(self.boxes[username])):
+                    if self.boxes[username][message].message_details['status'] == 'not read':
+                        self.boxes[username][message].message_details['status'] == 'read'
+                        count_message = count_message + 1
+                        return_message.append(self.boxes[username][message])
+                    if count_message == N:
+                        break
+
+        for message in range(len(self.boxes[username])):
+            if self.boxes[username][message].message_details['status'] == 'not read':
+                self.boxes[username][message].message_details['status'] == 'read'
+                return_message.append(self.boxes[username][message])
+        return return_message
+
+    def search_inbox(self, username, sentence):
+        """
+        An function that gets the name of the user and string .
+        the function return list of all the messages that contains the string in there headline or font.
+        :param username:  the name of the user
+        :param sentence:  string
+        :return: list of all the messages that contains the string in there headline or font.
+        """
+        return_message = []
+        for message in self.boxes[username]:
```

Comment:
Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features